rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ui/shaders/ShaderWidget.kt

```diff
@@ -0,0 +1,146 @@
+package com.jetbrains.rider.plugins.unity.ui.shaders
+
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.actionSystem.impl.SimpleDataContext
+import com.intellij.openapi.editor.impl.EditorImpl
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.rd.createNestedDisposable
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.wm.CustomStatusBarWidget
+import com.intellij.openapi.wm.StatusBarWidget
+import com.intellij.openapi.wm.StatusBarWidget.Multiframe
+import com.intellij.openapi.wm.impl.status.EditorBasedWidget
+import com.jetbrains.rd.platform.util.lifetime
+import com.jetbrains.rd.util.lifetime.Lifetime
+import com.jetbrains.rd.util.lifetime.SequentialLifetimes
+import com.jetbrains.rdclient.document.getFirstEditableEntityId
+import com.jetbrains.rider.UnityProjectDiscoverer
+import com.jetbrains.rider.cpp.fileType.CppFileType
+import com.jetbrains.rider.model.EditableEntityId
+import com.jetbrains.rider.model.ShaderContextData
+import com.jetbrains.rider.model.ShaderContextDataBase
+import com.jetbrains.rider.plugins.unity.UnityHost
+import icons.UnityIcons
+import java.awt.BorderLayout
+import java.awt.event.MouseEvent
+import java.awt.event.MouseListener
+import javax.swing.JComponent
+import javax.swing.JLabel
+import javax.swing.JPanel
+
+
+class ShaderWidget(project: Project) : EditorBasedWidget(project), CustomStatusBarWidget, Multiframe {
+
+    private val statusBarComponent = JPanel(BorderLayout())
+    private val label = JLabel(UnityIcons.FileTypes.ShaderLab)
+    private val requestLifetime = SequentialLifetimes(project.lifetime)
+
+    companion object {
+        fun getContextPresentation(data : ShaderContextData) = "${data.name}:${data.startLine}";
+    }
+
+    init {
+        label.text = "..."
+        statusBarComponent.isVisible = false
+        statusBarComponent.add(label)
+        statusBarComponent.addMouseListener(object : MouseListener {
+            override fun mouseClicked(e: MouseEvent?) {
+            }
+
+            override fun mousePressed(e: MouseEvent?) {
+            }
+
+            override fun mouseReleased(e: MouseEvent?) {
+                showPopup(label)
+            }
+
+            override fun mouseEntered(e: MouseEvent?) {
+            }
+
+            override fun mouseExited(e: MouseEvent?) {
+            }
+
+        })
+
+        // do nothing for not unity projects
+        if (UnityProjectDiscoverer.getInstance(project).isUnityProject) {
+            project.messageBus.connect(project.lifetime.createNestedDisposable())
+                .subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, object : FileEditorManagerListener {
```

Comment:
`EditorBasedWidget` already subscribes to the message bus, so you can just override `EditorBasedWidget.selectionChanged`